internal/rdc/client.go

```diff
@@ -175,8 +182,17 @@ func (c *Client) PollJob(ctx context.Context, id string, interval, timeout time.
 	}
 }
 
-func doRequestStatus(httpClient *http.Client, request *http.Request) (job.Job, error) {
-	resp, err := httpClient.Do(request)
+func doRequestStatus(httpClient *retryablehttp.Client, request *http.Request, interval time.Duration) (job.Job, error) {
+	httpClient.Logger = &logger.Logger{}
```

Comment:
I don't think we should let the client log. That's too verbose and the output won't tell the user much (or in some cases too much 😅 ).

What I meant by adding logs, is that we should log out and explain why something failed. Try putting yourself into the user's shoes. The job fails with the error message `failed to retrieve job status for suite` without any further explanation as to why. Would you be content? What if it starts failing all the time all of a sudden and you need to reach out to support, except they'll be just as helpless since the error message is not detailed enough to help with the investigation. This explanation should always happen, not just when verbose logging is on.